packages/web/src/pages/MainContentContext.tsx

```diff
@@ -2,28 +2,39 @@ import {
   createContext,
   memo,
   useRef,
-  MutableRefObject,
-  useContext
+  useContext,
+  useState,
+  MutableRefObject
 } from 'react'
 
 export const MAIN_CONTENT_ID = 'mainContent'
 
 export const MainContentContext = createContext({
-  mainContentRef: {} as MutableRefObject<HTMLDivElement | undefined>
+  setRef: (() => {}) as (node: HTMLDivElement) => void,
+  ref: {} as MutableRefObject<HTMLDivElement | undefined>,
+  ready: false
 })
 
 export const useMainContentRef = () => {
-  const { mainContentRef } = useContext(MainContentContext)
-  return mainContentRef
+  const { ref } = useContext(MainContentContext)
+  return ref
 }
 
 export const MainContentContextProvider = memo(
   (props: { children: JSX.Element }) => {
-    const mainContentRef = useRef<HTMLDivElement>()
+    const ref = useRef<HTMLDivElement>()
+    const [ready, setReady] = useState(false)
     return (
       <MainContentContext.Provider
         value={{
-          mainContentRef
+          setRef: (node: HTMLDivElement) => {
+            if (node && !ref.current) {
+              ref.current = node
+              setReady(true)
```

Comment:
Good stuff! Could maybe do the setstate in the useMainContentRef?

Reply:
interesting! how would that work? the hook might not get called again to set that state?